### PR TITLE
Implement prettier-ignore inside JSX

### LINKED ISSUE #2467: prettier-ignore doesn't work inside jsx
Using prettier-ignore is not supported inside jsx.
We have some compile issues (third-party compile issues), when prettier reformats some elements.

Example
```js
<div>
  {/* prettier-ignore */}
  <style jsx global>{ComponentStyles}</style>
</div>
```

outputs
```js
<div>
  {/* prettier-ignore */}
  <style jsx global>
    {ComponentStyles}
  </style>
</div>
```

### PR DESCRIPTION
**Pull request by azz**
Closes #2467

### CODE REVIEW

File: tests/jsx_ignore/jsfmt.spec.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+run_spec(__dirname);
+// run_spec(__dirname, null, ["babylon", "typescript"]);
```
Comment (vjeux) [inline tests/jsx_ignore/jsfmt.spec.js]: Why?

Comment (azz) [inline tests/jsx_ignore/jsfmt.spec.js]: Oops, forgot to switch that back, it's easier for debugging.

**Comment by azz** [comment]
Having not worked with the JSX AST much I don't know if I've missed any cases.

**Comment by azz** [comment]
Lets merge this one now and we can always make some adjustments later if I've missed things.

**Pull request merged**